client/a8c-for-agencies/sections/migrations/primary/self-migration-tool/index.tsx

```diff
@@ -17,15 +18,23 @@ const SelfMigrationTool = ( { type }: { type: 'pressable' | 'wpcom' } ) => {
 	const translate = useTranslate();
 
 	const stepInfo = getMigrationInfo( type, translate );
+	const { data: tipaltiData } = useGetTipaltiPayee();
 
 	const { pageTitle, heading, pageHeading, pageSubheading, steps, sessionStorageKey } = stepInfo;
 
-	const stepsWithCompletion = steps.map( ( step ) => {
-		return {
-			...step,
-			isCompleted: false,
-		};
-	} );
+	const stepsWithCompletion = steps
+		.map( ( step ) => {
+			return {
+				...step,
+				isCompleted: false,
+			};
+		} )
+		.filter( ( step ) => {
+			if ( tipaltiData?.Status === 'Active' && step.stepId === 'add-bank-info' ) {
+				return false;
+			}
+			return true;
+		} );
```

Comment:
- It is better to use `tipaltiData.IsPayable` instead of the status
- I think we could improve this using `reduce`

```suggestion
	const stepsWithCompletion = steps.reduce( ( acc, step ) => {
		if ( tipaltiData.IsPayable && step.stepId === 'add-bank-info' ) {
			return acc;
		}
		acc.push( {
			...step,
			isCompleted: false,
		} );
		return acc;
	}, [] );
```

Reply:
Good point. Thanks!